src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3413,6 +3413,8 @@ protected virtual void MarkMethodBody (MethodBody body)
 				return;
 			}
 
+			// Note: we need to mark compiler-generated method bodies in case they are only accessed via
+			// reflection, in which case we will never do a full data-flow scan of the user-defined method.
```

Comment:
Took me a while what this comment was talking about.
It sounds like we're marking the CG methods only if they're reflection accessed - which is not true, we mark them always, but for reflection-access-only methods we don't do dataflow.

Maybe:
"Note: we mark the method body of every method here including compiler generated methods whichever way they're accessed. But for compiler generated methods we only do data flow if they're used through their respective user method, we will skip data flow for reflection-access-only compiler generated methods below."